File: GestionEvenementBack/src/main/java/tn/eesprit/gestionevenementback/Entities/Transport.java
package tn.eesprit.gestionevenementback.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import lombok.experimental.FieldDefaults;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Transport implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer transportId;
    String startAdress;
    Float price;
    Integer capacity;
    @Enumerated(EnumType.STRING)
    TransportType transportType;
    @ManyToOne
    Event event;
    @OneToMany(mappedBy = "transport")
    Set<Reservation> passangers;


}

File: GestionEvenementFront/src/app/_Models/Transport.ts
import {TransportType} from "./TransportType";
import {Ordre} from "./Ordre";
import {Reservation} from "./Reservation";

export class Transport {
  transportId:string;
  fromAdress:string ;
  price:number ;
  capacity:number;
  transportType:TransportType ;
  event:Event;
  passangers:Reservation[];
}
